Tests/MnemonicSwiftTests.swift

```diff
@@ -1,156 +1,258 @@
+        let chineseMnemonic = "亂 語"
 
-    XCTAssertFalse(Mnemonic.validate(mnemonic: englishMnemonic))
-    XCTAssertFalse(Mnemonic.validate(mnemonic: chineseMnemonic))
-  }
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: englishMnemonic))
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: chineseMnemonic))
+    }
 
-  /// Test the empty string is determined to be an invalid mnemonic.
-  func testEmptyStringValidation() {
-    XCTAssertFalse(Mnemonic.validate(mnemonic: ""))
-  }
+    /// Test the empty string is determined to be an invalid mnemonic.
+    func testEmptyStringValidation() {
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: ""))
+    }
 
-  /// Test that strings in an unknown language are determined to be invalid.
-  func testUnknownLanguageValidation() {
-    let spanishMnemonic =
+    /// Test that strings in an unknown language are determined to be invalid.
+    func testUnknownLanguageValidation() {
+        let spanishMnemonic =
         "pera campesina pelican pen pera campesina pelican pen pera campesina pelican pen pera campesina pelican pen"
-    XCTAssertFalse(Mnemonic.validate(mnemonic: spanishMnemonic))
-  }
-
-  /// Test that strings of mixed case are determined to be valid.
-  func testMixedCaseValidation() {
-    let mixedCaseMnemonic = "pear PEASANT PeLiCaN pen"
-    XCTAssertTrue(Mnemonic.validate(mnemonic: mixedCaseMnemonic))
-  }
-
-  /// Test mixed language mnemonics.
-  func testMixedLanguageMnemonicValidation() {
-    let mixedLanguageMnemonic = "pear peasant pelican pen 路 级 少 图"
-    XCTAssertFalse(Mnemonic.validate(mnemonic: mixedLanguageMnemonic))
-  }
-
-  /// Test that strings padded with whitespace are determined to be valid.
-  func testWhitespacePaddedValidation() {
-    let whitespacePaddedMnemonic = "    pear peasant pelican pen\t\t\n"
-    XCTAssertTrue(Mnemonic.validate(mnemonic: whitespacePaddedMnemonic))
-  }
-
-  /// Test an valid mnemonic generates a seed string.
-  func testDeterministicSeedStringValidMnemonic() {
-    let invalidMnemonic =
-      "pear peasant pelican pen pear peasant pelican pen pear peasant pelican pen pear peasant pelican pen"
-    XCTAssertNotNil(Mnemonic.deterministicSeedString(from: invalidMnemonic))
-  }
-
-  /// Test an invalid mnemonic does not generate a seed string.
-  func testDeterministicSeedStringInvalidMnemonic() {
-    let invalidMnemonic =
-      "MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift MnemonicSwift"
-    XCTAssertNil(Mnemonic.deterministicSeedString(from: invalidMnemonic))
-  }
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: spanishMnemonic))
+    }
+
+    /// Test that strings of mixed case are determined to be valid.
+    func testMixedCaseValidation() {
+        let mixedCaseMnemonic = "pear PEASANT PeLiCaN pen"
+        XCTAssertThrowsError(try Mnemonic.validate(mnemonic: mixedCaseMnemonic))
+    }
```

Comment:
We should also test that it throws an exception for a seed phrase that's valid except for the only thing wrong with it is it has some uppercased character:

```
let x = "human pulse approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
let y = "Human pulse approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
XCTAssertNoThrow(try Mnemonic.validate(mnemonic: x)); // having this nearby makes it easy to look at the code and know exactly whats wrong with y
XCTAssertThrowsError(try Mnemonic.validate(mnemonic: y));
```
And also check that the high-level function fails too (just to make sure it's really calling validate):

```
XCTAssertNotNil(Mnemonic.deterministicSeedBytes(mnemonic: x));
XCTAssertNil(Mnemonic.deterministicSeedBytes(mnemonic: y));
```

And also verify the checksum catches word swaps:

```
let z = "pulse human approve subway climb stairs mind gentle raccoon warfare fog roast sponsor under absorb spirit hurdle animal original honey owner upper empower describe"
XCTAssertNil(Mnemonic.deterministicSeedBytes(mnemonic: z));
```